src/image/cache/minio-s3/index.ts

```diff
@@ -0,0 +1,60 @@
+import ImageCache from '../abstract'
+import { createHash } from 'crypto'
+import { Client } from 'minio'
+
+export default class MinioS3ImageCache extends ImageCache {
+  private static client: Client
+
+  public constructor (config, req) {
+    super(config, req)
+    if (MinioS3ImageCache.client === undefined) {
+      MinioS3ImageCache.client = new Client(this.moduleConfig.libraryOptions)
+    }
+  }
+
+  public get moduleConfig (): any {
+    return this.config.imageable.caching[`minio-s3`]
+  }
+
+  public async getImageFromCache () {
+    await new Promise((resolve, reject) => {
```

Comment:
`await new Promise` looks weird. Are you sure it works as supposed to?

Reply:
I would recommend to use `return new Promise...` here

Reply:
@revlis-x Please just return Promise it will be simpler
```
  public getImageFromCache () {
    return new Promise((resolve, reject) => {
```

Reply:
you're right!